### PR TITLE
Send requests on multiple server if it fails

### PR DESCRIPTION
**Pull request by luppiale**
This modification allows to send a request to an array of servers, this simplify the problem of contacting multiple servers when the connection fails for any reason. If every request fails you get the answer from the last server contacted.

Example:
Multiple replica server [host1.abc.com, host2.abc.com, host3.abc.com]
If the request on host1.abc.com of a resource fails then it will try contacting host2.abc.com, then host3.abc.com

### CODE REVIEW

**Comment by silkimen** [comment]
I don't think this should be handled in this plugin. You are free to write a wrapper around this plugin to solve this issue, because it solves a very special use-case, which probable won't meet the expectations of other users. e.g.
* person A could expect to receive an array of error responses and success responses for each request, instead of one response only
* another person B could expect that all requests are cancelled when one request fails

Basically it's the same class of problem like promises, you have following three builtin solutions:
* single promise
* multiple promises, all need to fulfill `Promise.all()`
* multiple promises, only one needs to fulfill `Promise.race()` <-- that's what you are expecting

So my suggestion is that you write a wrapper for your own use-case. You can even wrap it into a promise API (see above examples).
